src/shared/components/case-editor/services/wizard-page-field-to-case-field.mapper.ts

```diff
@@ -82,7 +84,34 @@ export class WizardPageFieldToCaseFieldMapper {
     }
   }
 
+  private hideParentIfAllChildrenHidden(caseField: CaseField) {
+    let children = [];
+    if (this.isCollection(caseField)) {
+      children = caseField.field_type.collection_field_type.complex_fields || [];
+    } else if (this.isComplex(caseField)) {
+      children = caseField.field_type.complex_fields || [];
+    }
+
+    children.forEach(e => this.hideParentIfAllChildrenHidden(e));
+
+    if (children.length > 0 && this.allCaseFieldsHidden(children)) {
+      caseField.hidden = true;
+    }
+  }
+
+  private allCaseFieldsHidden(children: CaseField[]): boolean {
+    return !children.some(e => e.hidden !== true);
+  }
+
   private isCollectionOfComplex(case_field: CaseField) {
     return case_field.field_type.type === 'Collection' && case_field.field_type.collection_field_type.type === 'Complex';
   }
+
+  private isComplex(case_field: CaseField) {
```

Comment:
Would have been good if these functions were on the model itself, if not at least in FieldsUtils to make them reusable.

Reply:
Utility methods are on https://tools.hmcts.net/jira/browse/RDM-4073 waiting to be merged